source/beamline/BioXAS/BioXASSideM1MirrorBendControl.cpp

```diff
@@ -0,0 +1,42 @@
+#include "BioXASSideM1MirrorBendControl.h"
+
+BioXASSideM1MirrorBendControl::BioXASSideM1MirrorBendControl(const QString &name, const QString &units, QObject *parent, const QString &description) :
+	BioXASMirrorBendControl(name, units, parent, description)
+{
+	// Initialize inherited variables.
+
+	setMinimumValue( calculateBendRadius(12, 12) );
+	setMaximumValue( calculateBendRadius(0.0001, 0.0001) );
+	setTolerance(100);
+
+	// Current settings.
+
+	updateStates();
+}
+
+BioXASSideM1MirrorBendControl::~BioXASSideM1MirrorBendControl()
+{
+
+}
+
+double BioXASSideM1MirrorBendControl::calculateBendRadius(double upstreamBenderValue, double downstreamBenderValue)
+{
+	double radius1 = -18654.3 + 144742/upstreamBenderValue + 5413.6 * log(upstreamBenderValue);
+	double radius2 = -28996.3 + 152538/downstreamBenderValue + 9534.08 * log(downstreamBenderValue);
+
+	double radius = (radius1 + radius2) / 2.0;
+
+	return radius;
+}
+
+double BioXASSideM1MirrorBendControl::calculateUpstreamBenderValue(double bendRadius)
+{
+	double result = 9.60504 + 62480.3/bendRadius - 0.797077 * log(bendRadius);
```

Comment:
Yup, what's needed for the equations is ln and that's what math.h 'log' gives us.